### PR TITLE
Fundamentals block development - landing and first pages

### PR DESCRIPTION
**Pull request by juanmaguitar**
This section provides an introduction to the most important concepts in Block Development.
This PR is related to the issue https://github.com/WordPress/gutenberg/issues/56282 and it publishes a first landing page with some initial content (https://github.com/WordPress/gutenberg/pull/56553, https://github.com/WordPress/gutenberg/pull/56551 and https://github.com/WordPress/gutenberg/pull/56334) that will be incremented with the new content planned for this sections as it gets published.

This PR also removes the whole [How to use JavaScript with the Block Editor](https://developer.wordpress.org/block-editor/how-to-guides/javascript/) as the most important ideas on this topic are covered at https://github.com/WordPress/gutenberg/pull/56553 

After the merge of this PR the following redirections should be implemented:

**Regarding Block Registration**
- [ ] From https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#block-registration to https://developer.wordpress.org/block-editor/getting-started/fundamentals-block-development/registration-of-a-block
- [ ] From https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#php-server-side to https://developer.wordpress.org/block-editor/getting-started/fundamentals-block-development/registration-of-a-block#registration-of-the-block-with-php-server-side
- [ ] From https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#javascript-client-side to https://developer.wordpress.org/block-editor/getting-started/fundamentals-block-development/registration-of-a-block#registration-of-the-block-with-javascript-client-side

**Regarding Javascript**
- [ ] From https://developer.wordpress.org/block-editor/how-to-guides/javascript to https://developer.wordpress.org/block-editor/getting-started/fundamentals-block-development/javascript-in-the-block-editor

### CODE REVIEW

**Comment by ndiego** [comment]
I think we should break this into two PRs, one for the new Landing page and another to remove the JavaScript section. We will need to create redirects for all the JavaScript pages, so we probably should only remove that once the new redirects are in place.

File: docs/getting-started/fundamentals-block-development/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Fundamentals of Block Development
+
+This section provides an introduction to the most important concepts in Block Development.
+
+In this section you will learn:
+
+1. [**File structure of a block**](https://developer.wordpress.org/block-editor/getting-started/fundamentals-block-development/file-structure-of-a-block) - The purpose of each one of the types of files available for a block, the relationships between them and their role in the output of the block
+<!--
```
Comment (ndiego) [inline docs/getting-started/fundamentals-block-development/README.md]: I am not sure we should include commented-out sections.

File: docs/getting-started/fundamentals-block-development/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+4. [**Storing data in block's attributes**](#) - How attributes are the native way for blocks to store data in them and the format they're stored in the DB.
+5. [**The block in the editor**](#) - How is defined the display and behaviour of the block in the editor from a developer's perspective.
+6. [**Using built-in components**](#) - How to use some common built-in components in Block Development
+7. [**Static or Dynamic rendering of a block**](#) - How blocks can have static (save time) or dynamic (request time) render for the frontend.
+8. [**Markup's representation of a block**](#) - How a block is represented in the DB using a specific HTML grammar.
+9. [**Styling blocks**](#) - The different methods available to define the styles of a block.
+10. [**Lifecycle of the block**](#) - The different cycles a block is involved in (load, save, update, render) and the different formats representing the block in these cycles.
+-->
+1. [**Javascript in the Block Editor**](https://developer.wordpress.org/block-editor/getting-started/fundamentals-block-development/javascript-in-the-block-editor) -  How to work with Javascript for the Block editor.
```
Comment (ndiego) [inline docs/getting-started/fundamentals-block-development/README.md]: Minor, but some lines end in a period and some don't.

**Comment by ndiego** [comment]
I also just noticed this, but should the URL be `/fundamentals-of-block-development` instead of `/fundamentals-block-development`?

**Comment by juanmaguitar** [comment]
@ndiego Thanks for the feedback!
I have addressed it and updated the PR

**Comment by juanmaguitar** [comment]
> I also just noticed this, but should the URL be `/fundamentals-of-block-development` instead of `/fundamentals-block-development`?

Would there be any specific benefit changing to `/fundamentals-of-block-development` the current URL (`/fundamentals-block-development`)

**Comment by ndiego** [comment]
> Would there be any specific benefit changing to `/fundamentals-of-block-development` the current URL (`/fundamentals-block-development`)

It just more accurately reflects the title of the page, but it makes the URL even longer. I wonder if just `/fundamentals` would work. It would be an easier URL to type/remember? I don't have a strong opinion either way.

**Comment by juanmaguitar** [comment]
> It just more accurately reflects the title of the page, but it makes the URL even longer. I wonder if just `/fundamentals` would work. It would be an easier URL to type/remember? I don't have a strong opinion either way.

@ndiego I have changed the folder to `fundamentals` and the references to it

**Review by ndiego — APPROVED** [review]
Looks good!

**Pull request merged**